tests/backup/backup_restore_basic_test.go

```diff
@@ -3189,3 +3190,145 @@ var _ = Describe("{DeleteNSDeleteClusterRestore}", func() {
+			nfsBackupLocationName = fmt.Sprintf("%s-%s-bl-%v", "nfs", getGlobalBucketName("nfs"), RandomString(4))
+			nfsBackupLocationUID = uuid.New()
+			backupLocationMap[nfsBackupLocationUID] = nfsBackupLocationName
+			err = CreateNFSBackupLocation(nfsBackupLocationName, nfsBackupLocationUID, orgID, " ", getGlobalBucketName("nfs"), true)
+			dash.VerifyFatal(err, nil, "Creating backup location")
+			log.InfoD("Creating AWS cred and S3 backup location")
+			s3CloudCredName = fmt.Sprintf("%s-%s-%v", "cred", "s3", RandomString(4))
+			s3BackupLocationName = fmt.Sprintf("%s-%s-bl-%v", "s3", getGlobalBucketName("aws"), RandomString(4))
+			s3CloudCredUID = uuid.New()
+			s3BackupLocationUID = uuid.New()
+			backupLocationMap[s3BackupLocationUID] = s3BackupLocationName
+			err = CreateCloudCredential("aws", s3CloudCredName, s3CloudCredUID, orgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] with [%s] as provider", s3CloudCredName, orgID, "aws"))
+			err = CreateS3BackupLocation(s3BackupLocationName, s3BackupLocationUID, s3CloudCredName, s3CloudCredUID, getGlobalBucketName("aws"), orgID, "")
+		})
+
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateApplicationClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			log.InfoD("Verifying cluster status for both source and destination clusters")
+			clusterStatus, err := Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			sourceClusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+			clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, destinationClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", destinationClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", destinationClusterName))
+		})
+
+		Step("Taking alternate backups of application from source cluster to both S3 and NFS backup locations", func() {
+			log.InfoD("Taking alternate backups of application from source cluster to both S3 and NFS backup locations")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, bkpNamespaces)
+			for i := 0; i < numberOfAlternateBackups; i++ {
+				for locationUID, locationName := range backupLocationMap {
```

Comment:
To ensure a consistent backup order between NFS and S3 with NFS first, using a map may be unreliable because it does not guarantee ordering. How about using a slice instead, such as `["nfs-uid": "nfs-name", "aws-uid": "aws-name"]`?

https://go.dev/play/p/AFnl6kguXLN

Reply:
Order of backup should not matter here I guess. Infact it is better if it is random. So that both the scenarios get tested.

Reply:
Thank you Mithun

Reply:
@kphalgun-px I think the objective is to take backup sequentially in alternating backup location so that the full backup taken on s3 doesn't have its incremental backup on NFS backup location. Parallelizing this will not give us that assurance.

Reply:
I mean, take a full backup and parallelize the incremental backups. But, anyways, it does not matter much.